lib/asciidoctor/reader.rb

```diff
@@ -835,10 +835,9 @@ def preprocess_include raw_target, raw_attributes
         warn %(asciidoctor: ERROR: #{line_info}: maximum include depth of #{@maxdepth[:rel]} exceeded)
         return false
       end
-      if ::RUBY_ENGINE_OPAL
+      if ::RUBY_ENGINE_OPAL && @document.attributes.has_key? 'env-browser'
```

Comment:
Could we also check if the global variable `JAVASCRIPT_PLATFORM` matches `browser` ? https://github.com/asciidoctor/asciidoctor.js/pull/248/files#diff-c3680ed68dbcf2cc0b36fa1af7000debR62

By default, I think Asciidoctor should choose the "right way" to process include files (and if it fails, users can always set the "env-browser" attribute). Wdyt ?